### DISCUSSION TITLE
Express and Typescript. Whats is the best practice to pass Prisma in the Request?

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#3716] [Q&A] [answered]
I am using ExpressJs.
I need to use `prisma` in all my routes. 
I have been thinking to  pass `Prisma` in the `Req`. In all routes, I will be able to access to prisma with `req.prisma.object.findOne()`

```ts
// index.ts
import express from 'express'
import { Request, Response, NextFunction } from 'express'
import { PrismaClient } from '@prisma/client'

const app = express()
const port = 9000

const stripeAuthorization = require('./routes/stripeAuthorization')
// declare module 'express-serve-static-core' {
//   interface Request {
//     prisma: PrismaClient
//   }
// }

app.use((req: Request, res: Response, next: NextFunction) => {
  const prisma = new PrismaClient()
  req.prisma = prisma
  next()
})

app.use('/', stripeAuthorization)

app.listen(port, () => console.log(`Example app listening on port ${port}!`))

```

It drops the error: `Property 'prisma' does not exist on type 'Request<ParamsDictionary, any, any, ParsedQs>'.ts(2339)`

![image](https://user-images.githubusercontent.com/15246526/93734387-efc75500-fb8d-11ea-8178-39f374004c49.png)

One quick around is by adding 

```ts
declare module 'express-serve-static-core' {
  interface Request {
    prisma: PrismaClient
  }
}
```

What solution do you recommend?
Thanks
Alan

**Comment by ryands17** [comment, marked as answer]
Hey @alan345 👋 
Yes, this error you're seeing is due to TS complaining as Prisma was not added on the `req` object in its type. So the best way to go would be what you have added here, is to add `PrismaClient` in the Express declarations.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
